### PR TITLE
undo (ctrl + Z)

### PR DESCRIPTION
**Pull request by arnon-1**
Very basic undo operation. (pressing ctrl + Z will undo the last few operations)
Advantages:
-Very simple
-Better than nothing (probably)
Disadvantages:
-Pressing ctrl + Z will only work once (pressing it again will result in the previous ctrl + Z press being undone). But the code can easily be expanded if needed.
-Not thoroughly tested at the moment, but I don't think I made any mistakes as the change is quite small.

### CODE REVIEW

**Comment by arnon-1** [comment]
Will resolve #107

File: web/scripts/app.js [hunk 1]
```diff
@@ -1297,8 +1311,18 @@ export class ComfyApp {
...
+		this.changing = false  // only used for administration for saved values for ctrl + Z
+
 		// Save current workflow automatically
-		setInterval(() => localStorage.setItem("workflow", JSON.stringify(this.graph.serialize())), 1000);
+		setInterval(function(){
+			const previous_workflow = localStorage.getItem("workflow")
+			const workflow = JSON.stringify(this.graph.serialize())
+			if(previous_workflow !== workflow){
+				if(!this.changing) localStorage.setItem("ctrlZ", previous_workflow);  // Save workflow for loading at ctrl + Z
+				this.changing = truep
```
Comment (jn-jairo) [inline web/scripts/app.js]: typo truep -> true

Comment (arnon-1) [reply web/scripts/app.js]: Ok, for some reason it also worked fine with the typo so I didn't notice it XD. Thanks for pointing it out!

**Review by jn-jairo — APPROVED** [review]

**Comment by jn-jairo** [comment]
It could store the previous X different workflows and that amount of changes can be a setting in the javascript UI, just save an array with the workflows and pop it on the ctrl + z

**Comment by arnon-1** [comment]
Yes, I did not implement that (yet) as I am not certain what would be the preferred method for this.
My idea was to keep saving until it exceeded a certain size (e.g. 1 Mb).
I am not sure if it would clog the UI if I would add an option for this as I think most users would never change the value.
So maybe a command line argument would be the best option?

Anyway, I think it is a 'strict' improvement at this moment, so I wanted to wait for this to get accepted before adding more functionality that might not get accepted.

**Pull request closed without merging**